.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,38 @@
+name: "CodeQL"
+on:
+  push:
```

Comment:
I actually haven't found codeql to be useful, it's never found an issue on the repos where I've used it, but it consumes an enormous amount of resources. If we include it at all I'd rather have it post-merge only on the cron.

Reply:
Sounds good. I will change it to merge.

Reply:
Removed to PR to main. Now it is merges to `main`